docs/api/sharedb-error.md

```diff
@@ -0,0 +1,98 @@
+---
+title: ShareDBError
+layout: default
+parent: API
+---
+
+# ShareDBError
+{: .no_toc }
+
+1. TOC
+{:toc}
+
+Representation of an error, with a machine-parsable [code](#error-codes).
+
+## Properties
+
+### `code` -- string
+
+> A machine-parsable [code](#error-codes) representing the type of error
+
+### `message` -- string
+
+> A human-readable message providing more detail about the error
+
+{: .warn }
+> Consumer code should never rely on the value of `message`, which may be fragile.
+
+## Error codes
+
+### `ERR_OP_SUBMIT_REJECTED`
+
+> The op submitted by the client has been rejected by the server for a non-critical reason.
+
+> When the client receives this code, it will attempt to roll back the rejected op, leaving the client in a usable state.
+
+> This error might be used as part of standard control flow. For example, consumers may define a middleware that validates document structure, and rejects operations that do not conform to this schema using this error code to reset the client to a valid state.
+
+### `ERR_OP_ALREADY_SUBMITTED`
+
+> The same op has been received by the server twice.
+
+> This is non-critical, and part of normal control flow, and is sent as an error in order to short-circuit the op processing. It is eventually swallowed by the server, and shouldn't need further handling.
+
+### `ERR_SUBMIT_TRANSFORM_OPS_NOT_FOUND`
+
+> The ops needed to transform the submitted op up to the current version of the snapshot could not be found.
+
+> If a client on an old version of a document submits an op, that op needs to be transformed by all the ops that have been applied to the document in the meantime. If the server cannot fetch these ops from the database, then this error is returned.
+
+> The most common case of this would be ops being deleted from the database. For example, let's assume we have a TTL set up on the ops in our database. Let's also say we have a client that is so old that the op corresponding to its version has been deleted by the TTL policy. If this client then attempts to submit an op, the server will not be able to find the ops required to transform the op to apply to the current version of the snapshot.
```

Comment:
```suggestion
> The most common case of this would be ops being deleted from the database. For example, let's assume we have a TTL set up on the ops in our database. Let's also say we have a client that is so out of date that the op corresponding to its version has been deleted by the TTL policy. If this client then attempts to submit an op, the server will not be able to find the ops required to transform the op to apply to the current version of the snapshot.
```